app/modules/sightings/resources.py

```diff
@@ -513,3 +515,55 @@ def delete(self, sighting):
         # TODO handle failure of feather deletion (when edm successful!)  out-of-sync == bad
         sighting.delete_cascade()
         return None
+
+
+@api.route('/<uuid:sighting_guid>/featured_asset_guid')
+@api.resolve_object_by_model(Sighting, 'sighting')
+class FeaturedAssetGuidBySightingID(Resource):
+    """
+    Featured Asset guid set and retrieval.
+    """
+
+    @api.login_required(oauth_scopes=['sightings:read'])
+    @api.permission_required(
+        permissions.ObjectAccessPermission,
+        kwargs_on_request=lambda kwargs: {
+            'obj': kwargs['sighting'],
+            'action': AccessOperation.READ,
+        },
+    )
+    def get(self, sighting):
+        """
+        Get featured Asset guid.
+        """
+        from app.modules.sightings.schemas import FeaturedAssetOnlySchema
+
+        asset_schema = FeaturedAssetOnlySchema()
+        return asset_schema.dump(sighting)
+
+    @api.login_required(oauth_scopes=['sightings:write'])
+    @api.permission_required(
+        permissions.ObjectAccessPermission,
+        kwargs_on_request=lambda kwargs: {
+            'obj': kwargs['sighting'],
+            'action': AccessOperation.WRITE,
+        },
+    )
+    def post(self, sighting):
+
+        request_in = json.loads(request.data)
+
+        context = api.commit_or_abort(
+            db.session,
+            default_error_message='Failed to update Sighting.featured_asset_guid.',
+        )
+        featured_asset_guid = request_in.get('featured_asset_guid', None)
+
+        success = False
+
+        if utils.is_valid_guid(featured_asset_guid):
+            sighting.set_featured_asset_guid(UUID(featured_asset_guid, version=4))
```

Comment:
rather than utils.is_valid_guid this could use @api.resolve_object_by_model(Asset, 'featured_asset') at the top of the method which would give you the real asset object, checking that it is one.

Reply:
I don't think I can do that.. We have already resolved the sighting below the route definition, we would need to `Asset.query.get` in order to access the featured_asset here.

Reply:
I'd like to get this in before I go out of town tomorrow, but If I'm wrong I'd be happy make a commit to change it when I'm back next week.

Reply:
Happy with this, I may have a play today to see if I'm right